platforms/core-runtime/gradle-cli-main/src/main/java/org/gradle/launcher/GradleMain.java

```diff
@@ -29,8 +28,6 @@ public static void main(String[] args) throws Exception {
             System.exit(1);
         }
 
-        Class<?> mainClass = Class.forName("org.gradle.launcher.bootstrap.ProcessBootstrap");
-        Method mainMethod = mainClass.getMethod("run", String.class, String[].class);
-        mainMethod.invoke(null, "org.gradle.launcher.Main", args);
+        ProcessBootstrap.run("org.gradle.launcher.Main", args);
```

Comment:
Don't need to use reflection any more.

Reply:
Does this class still live in `:launcher`? 

I would expect that `gradle-cli-main` would need a runtimeOnly dependency on that project

Reply:
It does still live in ":launcher". `ProcessBootstrap` goes and finds everything from the distribution it is running from and sets up the appropriate ClassLoader structure for running a build, rather than using its runtime classpath.

This allows everything to be properly isolated, and the runtime classpath for the "entry point" can just be the absolute minimum to start the process.

At some point, I'd like to try having the `gradle` command include the client statically and only do the discovery and ClassLoader stuff lazily if a build actually needs to run in the client process (and this can go away when `--no-daemon` goes away). I think this will help with, for example, creating a native client, etc.